File: Entities/UserEntity.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace BerryNameApi.Entities
{
    public class UserEntity
    {
        [BsonId]
        [BsonRepresentation(BsonType.String)]
        public Guid Id { get; set; }

        public string Email { get; set; } = string.Empty;
        public string PasswordHashed { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string Address { get; set; } = string.Empty;

        [BsonRepresentation(BsonType.String)]
        public Role Role { get; set; } = Role.Worker;

        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

    }

    public enum Role
    {
        Boss,
        Worker
    }
}

File: entities/user_entity.py
from datetime import datetime
from pydantic import BaseModel


# 사용자 엔티티
class UserEntity(BaseModel):
    id: str
    name: str
    email: str
    password: str
    address: str
    role: str
    updated_at: datetime
    created_at: datetime

    def __str__(self):
        return f"UserEntity(id={self.id}, name={self.name}, email={self.email}, role={self.role}, created_at={self.created_at}, updated_at={self.updated_at})"
